### DISCUSSION TITLE
How to avoid longer streaks of a road

### DISCUSSION THREAD

**Discussion by Miq1** [osmandapp/OsmAnd#20135] [Q&A]
For some years there are construction works on a major motorway in Germany (A45), including closed bridges with detours etc.
There is a section of this motorway (between junctions to motorways A1 and A4) that best should be avoided for long-distance routings in favour to other motorways around it.

Same holds true for the A12 motorway in the Netherlands (between Venendaal and Utrecht).

Unfortunately the "Avoid road" tool will only mark short parts of that section, which in turn will lead Osmand+ to just do another local detour for it. One would have to mark literally dozens of segments as to be avoided.

Is there a better way to get Osmand+ to route around this large section?

**Comment by Miq1** [comment]
On a side note: there is a workaround - use route planning and place waypoints on the motorways you would prefer to be used, then navigate with that route as a base.

While this works, it is no viable solution for everyday's navigation.

**Comment by DmitryAlexei** [comment]
Have you tried "consider temporary limitations" option? see docs https://osmand.net/docs/user/navigation/routing/osmand-routing#consider-temporary-limitations + live updates?

**Reply by Miq1** [reply]
@DmitryAlexei of course I did and this is enabled. But even then there is no preference to have wider detours using other motorways, instead one is routed on country streets to get off an on the blocked motorway close to the interruption. I happen to know the German example quite well; in that region most streets off the motorway are windy, crammed with traffic and without alternatives nearby. It would be very handy to mark "avoid this road" and then have the option to "extend to next intersections with roads of the same category". If a motorway is blocked, that would effectively have me routed around using different motorways. It would be up to me if I chose to get that or the regular shorter detour.
